elastic/kibana
# Add execution context data to discover, vega and timelion

## Summary
part of https://github.com/elastic/kibana/issues/102629
### Discover
`discover` might exist in different containers:
For the saved searches on `dashboard`, we pass the standard execution context information.
For the searches initiated within `discover` app, the execution context contains limited information: we cannot provide a unique identifier nor a specific URL. Let me know if you have a better idea what we can (and should) provide additionally.

### Vega
Vega lost `executionContext` in the expression, fixed to manually propagate the object to the `search` function.

### Timelion
`executionContext`  hasn't been passed from the expression to an `http` call to an ednpoint initiating a request to ES. Fixed.

### How to test
```yml
logging:
  appenders:
    myconsole:
      type: console
      layout:
        type: pattern
        pattern: '%date|%meta'
  loggers:
    - name: elasticsearch.query
      level: all
      appenders: [myconsole]
    - name: elasticsearch.query.taskManager
      level: off
    - name: execution_context
      level: debug
      appenders: [console]
```
- run Kibana
- navigate to Kibana Browser App
- Install sample data. I tested on `Sample web logs` and `Sample eCommerce orders`
- Navigate to dashboards.
- You should be able to see the output like this:

Vega (on `web logs` dashboard):
```
[2021-08-10T11:42:31.657+03:00][DEBUG][execution_context] stored the execution context: {"parent":{"type":"application","name":"dashboard","id":"edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b","description":"[Logs] Web Traffic","url":"/view/edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b"},"type":"visualization","name":"Vega","id":"cb099a20-ea66-11eb-9425-113343a037e3","description":"[Logs] Unique Visitor Heatmap","url":"/app/visualize#/edit/cb099a20-ea66-11eb-9425-113343a037e3"}

2021-08-10T11:42:31.661+03:00|{"http":{"request":{"id":"c89d1ab3-b4a7-4920-a64a-22a910a413b0;kibana:application:dashboard:edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b;visualization:Vega:cb099a20-ea66-11eb-9425-113343a037e3"}}}
```

Discover (on `ecommerce orders` dashboard):
```
[2021-08-10T12:38:35.204+03:00][DEBUG][execution_context] stored the execution context: {"parent":{"type":"application","name":"dashboard","id":"722b74f0-b882-11e8-a6d9-e546fe2bba5f","description":"[eCommerce] Revenue Dashboard","url":"/view/722b74f0-b882-11e8-a6d9-e546fe2bba5f"},"type":"search","name":"discover","id":"3ba638e0-b894-11e8-a6d9-e546fe2bba5f","description":"[eCommerce] Orders","url":"/app/discover#/view/3ba638e0-b894-11e8-a6d9-e546fe2bba5f"}

2021-08-10T12:38:35.208+03:00|{"http":{"request":{"id":"8fd90dde-9290-4f7b-8181-0acd2b0a0b1d;kibana:application:dashboard:722b74f0-b882-11e8-a6d9-e546fe2bba5f;search:discover:3ba638e0-b894-11e8-a6d9-e546fe2bba5f"}}}
```
